pkg/geo/geogfn/segmentize.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"math"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/geo/geographiclib"
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/s2"
+	"github.com/twpayne/go-geom"
+)
+
+// Segmentize return modified Geography having no segment longer
+// that given maximum segment length.
+func Segmentize(geography *geo.Geography, segmentMaxLength float64) (*geo.Geography, error) {
+	spheroid := geographiclib.WGS84Spheroid
+	// Convert segmentMaxLength to s1.ChordAngle as further calculation
+	// is done considering segmentMaxLength as s1.ChordAngle.
+	segmentMaxLength = segmentMaxLength / spheroid.SphereRadius
```

Comment:
make this of type `s1.ChordAngle`:
`s1.ChordAngleFromAngle(s1.Angle(segmentMaxLength/spheroid.SphereRadius))`

and rename it to `segmentMaxChordAngle`

Reply:
This will be more efficient doing this way,
but I think `s1.ChordAngle` does not shows linear behaviour as
```
a := s1.ChordAngleFromAngle(s1.Angle(1))
b := s1.ChordAngleFromAngle(s1.Angle(0.5))

fmt.Println(a/b)
```
the expected ratio must be 2 but its actually 3.755165123780745
therefore, the ratio is not maintained after I convert them to `s1.ChordAngle`

Reply:
Good find!